bundles/org.openhab.core.persistence/src/main/java/org/openhab/core/persistence/internal/PersistenceManagerImpl.java

```diff
@@ -256,46 +261,50 @@ private boolean appliesToItem(SimpleItemConfiguration config, Item item) {
+                                PersistenceService service = persistenceServices.get(serviceName);
+                                if (service instanceof QueryablePersistenceService) {
+                                    QueryablePersistenceService queryService = (QueryablePersistenceService) service;
+                                    FilterCriteria filter = new FilterCriteria().setItemName(item.getName())
+                                            .setPageSize(1);
+                                    Iterable<HistoricItem> result = safeCaller
+                                            .create(queryService, QueryablePersistenceService.class).onTimeout(() -> {
+                                                logger.warn("Querying persistence service '{}' takes more than {}ms.",
+                                                        queryService.getId(), SafeCaller.DEFAULT_TIMEOUT);
+                                            }).onException(e -> {
+                                                logger.error(
+                                                        "Exception occurred while querying persistence service '{}': {}",
+                                                        queryService.getId(), e.getMessage(), e);
+                                            }).build().query(filter);
+                                    if (result != null) {
+                                        Iterator<HistoricItem> it = result.iterator();
+                                        if (it.hasNext()) {
+                                            HistoricItem historicItem = it.next();
+                                            GenericItem genericItem = (GenericItem) item;
+                                            genericItem.removeStateChangeListener(this);
+                                            genericItem.setState(historicItem.getState());
+                                            genericItem.addStateChangeListener(this);
+                                            if (logger.isDebugEnabled()) {
+                                                logger.debug("Restored item state from '{}' for item '{}' -> '{}'",
+                                                        DateFormat.getDateTimeInstance()
+                                                                .format(historicItem.getTimestamp()),
+                                                        item.getName(), historicItem.getState());
+                                            }
+                                            return;
                                         }
-                                        return;
                                     }
+                                } else if (service != null) {
+                                    logger.warn(
+                                            "Failed to restore item states as persistence service '{}' can not be queried.",
```

Comment:
Minor spelling improvement:

```suggestion
                                            "Failed to restore item states as persistence service '{}' cannot be queried.",
```